test/crit/Makefile

```diff
@@ -1,19 +1,29 @@
 CC ?= gcc
+GO ?= go
 
 all: test clean
 
-test: gen-imgs
+test: gen-imgs integration-test e2e-test
+
+integration-test:
+	@echo "Running integration test"
+	@./crit-test
+
+e2e-test:
```

Comment:
We need to add the corresponding dependency targets to be able to run the tests with
`sudo make -C test/crit integration-test` and `sudo make -C test/crit e2e-test`
```diff
--- a/test/crit/Makefile
+++ b/test/crit/Makefile
@@ -3,17 +3,17 @@ GO ?= go
 
 all: test clean
 
-test: gen-imgs integration-test e2e-test
+test: integration-test e2e-test
 
-integration-test:
+integration-test: gen-imgs crit-test
        @echo "Running integration test"
        @./crit-test
 
-e2e-test:
+e2e-test: gen-imgs
        @echo "Running E2E test"
        @./crit-test.sh
 
-gen-imgs: ../loop/loop crit-test
+gen-imgs: ../loop/loop
        $(eval PID := $(shell ../loop/loop))
        criu dump -v4 -o dump.log -D ./ -t $(PID)
```

Reply:
Adding `gen-imgs` as a dep for the individual test targets would mean that we'll be generating two sets of images, which would make things messy. Since the individual targets will be called only for debugging, I thought we can just run `gen-imgs` manually before that.

Reply:
To solve this problem we can make this a non-phony target.
For example:
```diff
--- a/test/crit/Makefile
+++ b/test/crit/Makefile
@@ -3,19 +3,20 @@ GO ?= go
 
 all: test clean
 
-test: gen-imgs integration-test e2e-test
+test: integration-test e2e-test
 
-integration-test:
+integration-test: test-images crit-test
        @echo "Running integration test"
        @./crit-test
 
-e2e-test:
+e2e-test: test-images
        @echo "Running E2E test"
        @./crit-test.sh
 
-gen-imgs: ../loop/loop crit-test
+test-images: ../loop/loop
        $(eval PID := $(shell ../loop/loop))
-       criu dump -v4 -o dump.log -D ./ -t $(PID)
+       mkdir -p $@
+       criu dump -v4 -o dump.log -D $@ -t $(PID)
 
 ../loop/loop: ../loop/loop.c
        $(CC) $^ -o $@
@@ -24,6 +25,6 @@ crit-test: main.go
        $(GO) build -v -o $@ $^
 
 clean:
-       @rm -f *.img *.log *.json stats-*
+       @rm -rf test-images
 
-.PHONY: all test integration-test e2e-test gen-imgs clean
+.PHONY: all test integration-test e2e-test clean
```
It creates a folder called `test-images` and this target would run only when the folder does not exist.